### PR TITLE
Panzer: Add a communicator parameter to FaceToElement methods

### LINKED ISSUE #10669: Panzer: FaceToElement::initialize() assumes MPI_COMM_WORLD
## Bug Report
@trilinos/panzer 

### Description
panzer::FaceToElement::initialize() uses MPI_COMM_WORLD even if the panzer::ConnectionManager was created with a sub-communicator.

### PR DESCRIPTION
**Pull request by tkordenbrock**
FaceToElement::initialize(ConnectionManager) was hardcoded to use MPI_COMM_WORLD.
 - Added a constructor and initialize() method to the FaceToElement class that
   has a communicator parameter.
 - For backward compatibility, the old constructor and initialize() method now
   pass MPI_COMM_WORLD as a parameter to new methods.
Added initialize() tests for default, explicit MPI_COMM_WORLD and sub-communicator.

@trilinos/panzer 

## Related Issues

* Closes #10669 

## Testing
A unit test was added to test initialization with a default and explicit communicators.

### CODE REVIEW

File: packages/panzer/disc-fe/src/Panzer_FaceToElement.hpp [hunk 1]
```diff
@@ -78,10 +78,19 @@ class FaceToElement {
 
   FaceToElement(panzer::ConnManager & conn);
 
-  /** Build the mapping from a mesh topology.
+  FaceToElement(panzer::ConnManager & conn,
+                const Teuchos::RCP<const Teuchos::Comm<int>> comm);
+
+  /** Build the mapping from a mesh topology using MPI_COMM_WORLD.
     */
   void initialize(panzer::ConnManager & conn);
```
Comment (eric-c-cyr) [inline packages/panzer/disc-fe/src/Panzer_FaceToElement.hpp:86]: Are you maintaining this for backwards compatiblity?

**Review by eric-c-cyr — APPROVED** [review]
Thanks for the change @tkordenbrock

File: packages/panzer/disc-fe/src/Panzer_FaceToElement.hpp (hunk 1 above)
Comment (eric-c-cyr) [reply packages/panzer/disc-fe/src/Panzer_FaceToElement.hpp:86]: Lets do that, just to keep things clean.

Comment (tkordenbrock) [reply packages/panzer/disc-fe/src/Panzer_FaceToElement.hpp:86]: OK.  Since it looks like the autotester is backed up, I will deprecate it now.

**Review by eric-c-cyr — APPROVED** [review]
Great

**Pull request merged**